src/performance/manager.py

```
@@ -0,0 +1,182 @@
+        
+    def _group_by_day(self, metrics: List[Dict[str, Any]]) -> Dict[str, List[Dict[str, Any]]]:
+        """Group metrics by day"""
+        daily = {}
+        for metric in metrics:
+            day = metric["timestamp"].date().isoformat()
+            if day not in daily:
+                daily[day] = []
+            daily[day].append(metric)
+        return daily
+        
+    def _calculate_daily_averages(self, daily_metrics: Dict[str, List[Dict[str, Any]]]) -> Dict[str, Dict[str, float]]:
```

Ruff (0.8.2)

Use `dict` instead of `Dict` for type annotation

Replace with `dict`

(UP006)